main/lsp/requests/definition.cc

```diff
@@ -9,6 +11,26 @@ DefinitionTask::DefinitionTask(const LSPConfiguration &config, MessageId id,
                                unique_ptr<TextDocumentPositionParams> params)
     : LSPRequestTask(config, move(id), LSPMethod::TextDocumentDefinition), params(move(params)) {}
 
+core::Loc
+DefinitionTask::findRequireRelativeLoc(const core::GlobalState &gs,
+                                       const std::vector<std::unique_ptr<core::lsp::QueryResponse>> &responses) {
+    if (responses.size() < 2)
+        return core::Loc::none();
+
+    auto parentSendResp = move(responses[1])->isSend();
+    const bool isRequireRelative = parentSendResp && parentSendResp->callerSideName == core::Names::require_relative();
+    auto literal = move(responses[0])->isLiteral();
+    if (isRequireRelative && literal) {
+        auto literalValue = core::cast_type_nonnull<core::LiteralType>(literal->retType.type).asName(gs).shortName(gs);
+        auto relativeFileName = fmt::format("{}.rb", literalValue);
```

Comment:
This is going to be `../../some/relative/path.rb`, right?

Reply:
Also, I _think_ `require_relative` works if you explicitly specify the extension too...

Reply:
Correct for 1/, for 2/ I'll check but that seems plausible and this code should be made to support it